avro2tf/src/main/scala/com/linkedin/avro2tf/jobs/FeatureListGeneration.scala

```diff
@@ -160,46 +194,120 @@ class FeatureListGeneration {
    * @param params TensorizeIn parameters specified by user
    * @param fileSystem A file system
    */
-  private def writeFeatureList(params: TensorizeInParams, fileSystem: FileSystem): Unit = {
+  private def writeFeatureList(
+    params: TensorizeInParams,
+    fileSystem: FileSystem,
+    ntvTensors: mutable.HashSet[String]): Unit = {
+
+    /**
+     * Write feature list to disk for a list of (feature entry, count) pairs
+     *
+     * @param p Path to file to create and write
+     * @param featureEntriesWCount a list of (feature entry, count) pairs
+     */
+    def writeFeatureEntriesWCountToDisk(
+      p: Path,
+      featureEntriesWCount: Seq[(String, Int)],
+      prefix: Option[String]): Unit = {
+
+      val outputStream = fileSystem.create(p)
+      val writer = new OutputStreamWriter(outputStream, UTF_8.name())
+      prefix match {
+        case Some(prefixString)
+        => {
+          featureEntriesWCount.foreach {
+            case (featureEntry, _) => writer.write(s"$prefixString,$featureEntry\n")
+          }
+        }
+        case None => {
+          featureEntriesWCount.foreach {
+            case (featureEntry, _) => writer.write(s"$featureEntry\n")
+          }
+        }
+      }
+      writer.close()
+    }
 
-    val tmpFeatureListPath = s"${params.workingDir.rootPath}/$TMP_FEATURE_LIST"
-    var currentColumnName: String = EMPTY_STRING
-    var writer: OutputStreamWriter = null
-
-    fileSystem.globStatus(new Path(s"$tmpFeatureListPath/$FILE_NAME_REGEX"))
-      .map(fileStatus => fileStatus.getPath.toString)
-      .toSeq
-      .sortWith((path1, path2) => path1 < path2)
-      .foreach(
-        fileName => {
-          val inputStream = fileSystem.open(new Path(fileName))
-
-          scala.io.Source.fromInputStream(inputStream, UTF_8.name())
-            .getLines()
-            .foreach(
-              line => {
-                // Get column name of FeatureListEntry
-                val columnName = line.split(SPLIT_REGEX).head
-
-                // Reassign the current column name variable if column name is different
-                if (columnName != currentColumnName) {
-                  currentColumnName = columnName
-                  val outputStream = fileSystem.create(
-                    new Path(s"${params.workingDir.featureListPath}/$columnName")
-                  )
-
-                  if (writer != null) writer.close()
-                  writer = new OutputStreamWriter(outputStream, UTF_8.name())
-                }
-
-                // Get the feature entry of FeatureListEntry
-                val featureListEntry = line.stripPrefix(columnName + SPLIT_REGEX)
-                writer.write(s"$featureListEntry\n")
-              })
-          inputStream.close()
-        })
-
-    if (writer != null) writer.close()
-    fileSystem.delete(new Path(tmpFeatureListPath), ENABLE_RECURSIVE)
+    // first get a set of tensor names for which temporary feature list files have been collected
```

Comment:
can we wrap line 231-243 to a method to make current method a little modularized.

Reply:
Good point. Will address.